### PR TITLE
chore: remove overlay mining

### PR DESCRIPTION
**Pull request by vladopajic**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
Closes: #3254

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3684)

### CODE REVIEW

File: pkg/node/node.go [hunk 1]
```diff
@@ -491,45 +487,6 @@ func NewBee(ctx context.Context, addr string, publicKey *ecdsa.PublicKey, signer
 	if err != nil {
 		return nil, fmt.Errorf("check presence of nonce: %w", err)
 	}
-	if !nonceExists {
```
Comment (istae) [inline pkg/node/node.go:494]: a fresh node may not have a nonce, so this is needed I think

Comment (istae) [reply pkg/node/node.go:494]: overlayNonceExists can be refactored to return an 32 byte zero slice in the case that an old nonce does not exist

**Review by istae — CHANGES_REQUESTED** [review]

File: pkg/node/node.go (hunk 1 above)
Comment (vladopajic) [reply pkg/node/node.go:494]: isn't this already the case?

Comment (istae) [reply pkg/node/node.go:494]: aha yes

**Review by istae — APPROVED** [review]

**Review by notanatol — APPROVED** [review]

**Pull request merged**